docs/content/release-notes/0.3.0.md

```diff
@@ -42,17 +61,83 @@ switch "test-switch" {
 
 ```
 
-Documentation was updated in [\#371](https://github.com/asteris-llc/converge/pull/371) so that links like
-[converge.aster.is/0.2.0](https://converge.aster.is/0.2.0) will resolve documentation for that version.
+#### Named Locks
+
+Basic locks have been added to the engine. This allows worksflows where multiple tasks run
+one at a time instead of in parallel. Locks are added as a node in the graph and released
+after all tasks sharing the lock are complete.
+
+For example, `apt` package installs cannot run in parallel. In the following example
+each task with the `apt` lock is run sequentually.
+
+```hcl
+task "install-jq" {
+  check = "dpkg -s jq >/dev/null 2>&1"
+  apply = "apt-get update 2>&1 > /dev/null && apt-get -y install jq"
+  lock  = "apt"
+}
+
+task "install-build-essential" {
+  check = "dpkg -s build-essential >/dev/null 2>&1"
+  apply = "apt-get update 2>&1 > /dev/null && apt-get -y install build-essential"
+  lock  = "apt"
+}
+```
+
+#### Node Metadata
+
+Initial support for adding metadata to graph nodes was added via [\#369](https://github.com/asteris-llc/converge/pull/369). 
```

Comment:
this is not a user facing change yet. It will be in the next release when we get better errors, names, etc out of it. But maybe it shouldn't be in the release notes?